#167899197 Users Should Be Able To Read Articles

JuwonAbiola opened the pull request:
#### What does this PR do?
- Enables a user to view an article
#### Description of Task proposed in this pull request?
- Create viewArticle method
- Update Swagger Docs
- Write test to test viewArticle method

#### How should this be manually tested (Quality Assurance)?
- Explain how to manually test this feature. For example if changes were made in
the UI or in the API, explain where and if any specific access is needed.

#### What are the relevant pivotal tracker stories?
- [167899197](https://www.pivotaltracker.com/story/show/167899197)

#### Any background context you want to add (Operations Impact)?
- The endpoint for view article is not protected 
- Run `npm run db:rollback` the article seeder file was updated..

#### What I have learned working on this feature: 
- I learned how to stub 500 error

#### Screenshots: 
N/A

chialuka commented on server/controllers/Articles.js:
```diff
@@ -335,6 +335,57 @@ const createLikeOrDislike = async (userAction, userId, article) => {
+      } = req;
+      const article = await Article.findBySlug(slug);
+      if (!article || article.isArchived) {
+        return serverResponse(res, 404, { error: 'article not found' });
```
Shouldn't a logged in user be able to view his own archived articles?

chialuka commented on test/articles/viewArticle.test.js:
```diff
@@ -0,0 +1,76 @@
+
+describe('GET Article', () => {
+  context('when a user views a published article', () => {
+    it('it returns the user profile', async () => {
```
There is an extra "it" in your it block

chialuka commented on test/articles/viewArticle.test.js:
```diff
@@ -0,0 +1,76 @@
+  });
+
+  context('when a user tries to view an article that does not exist', () => {
+    it('it returns a not found error', async () => {
```
ditto

chialuka commented on server/controllers/Articles.js:
```diff
@@ -335,6 +335,57 @@ const createLikeOrDislike = async (userAction, userId, article) => {
+      if (!article || article.isArchived) {
+        return serverResponse(res, 404, { error: 'article not found' });
+      }
+      const {
```
Is there a reason for de-structuring the article object?

ayodejiAA commented:
@jhayX, the return does not contain 'following'

JuwonAbiola committed:
feature[167899197]Users should be able to read articles
- create controller method to view article
- write test to test view article controller method
[Delivers #167899197]
